### PR TITLE
feat: give network manager access to negotiations

### PR DESCRIPTION
**Pull request by RadovanTomik**
### Description:

Modifies the system so that the Network Manager is able to read relevant negotiations as well as post comments. Since this required changes to multiple components some of them, such as Post Service had to be refactored.

### Checklist:

_Make sure you tick all the boxes below if they are true or do not apply before you ask for review_

- [x] I have performed a self-review of my code
- [x] I have made my code as simple as possible
- [x] I have added relevant tests for my changes and the code coverage has not dropped substantially
- [x] I have removed all commented code
- [x] I have updated the documentation in all relevant places (Javadoc, Swagger, MDs...)
- [x] I have described the PR and added a meaningful title in the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java [hunk 1]
```diff
@@ -26,6 +33,9 @@ public static Long getCurrentlyAuthenticatedUserInternalId() throws ClassCastExc
    * @return true or false
    */
   public static boolean isCurrentlyAuthenticatedUserAdmin() {
+    if (Objects.isNull(SecurityContextHolder.getContext().getAuthentication())) {
+      return false;
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java:37]:
```java
return !Objects.isNull(SecurityContextHolder.getContext().getAuthentication()) && SecurityContextHolder.getContext().getAuthentication().getAuthorities().stream()
        .anyMatch(a -> a.getAuthority().equals("ROLE_ADMIN"));
```

File: src/test/java/eu/bbmri_eric/negotiator/integration/api/v3/NegotiationControllerTests.java [hunk 2]
```diff
@@ -1212,22 +1212,13 @@ void getNegotiationsForAdmin_doesNotHaveRoleAdmin_Forbidden() throws Exception {
...
-
-  @Test
-  @WithMockUser(authorities = "ROLE_USER") // Assuming a non-admin user
+  @WithMockNegotiatorUser(id = 102L) // Assuming a non-admin user
   void sendEvent_NonAdmin_Forbidden() throws Exception {
     mockMvc
         .perform(
             MockMvcRequestBuilders.put(
                 "%s/negotiation-1/lifecycle/APPROVE".formatted(NEGOTIATIONS_URL)))
+        .andDo(print())
```
Comment (svituz) [inline src/test/java/eu/bbmri_eric/negotiator/integration/api/v3/NegotiationControllerTests.java:1221]: please remove debugging statements

File: src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java [hunk 3]
```diff
@@ -15,9 +17,14 @@ public class AuthenticatedUserContext {
...
    */
-  public static Long getCurrentlyAuthenticatedUserInternalId() throws ClassCastException {
-    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-    return ((NegotiatorPrincipal) auth.getPrincipal()).getPerson().getId();
+  public static Long getCurrentlyAuthenticatedUserInternalId()
+      throws AuthenticationCredentialsNotFoundException {
+    try {
+      Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+      return ((NegotiatorPrincipal) auth.getPrincipal()).getPerson().getId();
+    } catch (Exception e) {
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java:25]: please use the specific exception

File: src/main/java/eu/bbmri_eric/negotiator/governance/resource/NonRepresentedResourcesHandlerImpl.java
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/governance/resource/NonRepresentedResourcesHandlerImpl.java:1]: resourceLifecycleService is not used anymore. You can get rid of it in this class

File: src/main/java/eu/bbmri_eric/negotiator/governance/resource/NonRepresentedResourcesHandlerImpl.java [hunk 4]
```diff
@@ -37,31 +36,13 @@ public void updateResourceInOngoingNegotiations(Long resourceId, String sourceId
         continue;
       }
       if (Objects.equals(state, NegotiationResourceState.REPRESENTATIVE_UNREACHABLE)) {
-        updateResourceStatus(negotiation.getId(), sourceId);
+        updateResourceStatus(negotiation, sourceId);
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/governance/resource/NonRepresentedResourcesHandlerImpl.java:39]: Here you can directly call 
```java
negotiation.setStateForResource(sourceId, NegotiationResourceState.REPRESENTATIVE_CONTACTED);
```

see the following comment

File: src/main/java/eu/bbmri_eric/negotiator/governance/resource/ResourceServiceImpl.java
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/governance/resource/ResourceServiceImpl.java:1]: `requestRepository` and `userNotificationService` are not used. You can remove them from the class

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java [hunk 5]
```diff
@@ -262,6 +265,10 @@ public Iterable<NegotiationDTO> findByFiltersForUser(
   @Override
   public Iterable<NegotiationDTO> findAllForNetwork(
       Long networkId, NegotiationFilterDTO filtersDTO) {
+    if (!personRepository.isNetworkManager(
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java:268]: shouldn't this be delegated to the new NegotiationAccessManager?

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java [hunk 6]
```diff
@@ -301,8 +308,14 @@ private Negotiation findEntityById(String negotiationId, boolean includeDetails)
    * @param includeDetails whether the negotiation returned include details
    * @return the Negotiation with specified negotiationId
    */
+  @Override
   public NegotiationDTO findById(String negotiationId, boolean includeDetails)
       throws EntityNotFoundException {
+    Long userID = AuthenticatedUserContext.getCurrentlyAuthenticatedUserInternalId();
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java:314]: you can move this statement after the following `if`, so it is not called if unnecessary (really pedantic :) )

File: src/test/java/eu/bbmri_eric/negotiator/integration/api/v3/PostControllerTests.java [hunk 7]
```diff
@@ -109,17 +149,22 @@ public void testCreatePrivatePostWithUnknownResource() throws Exception {
...
         .andExpect(status().isOk())
         .andExpect(jsonPath("$._embedded.posts.length()", is(numberOfPosts)))
-        .andExpect(jsonPath("$._embedded.posts[0].id", is("post-1-researcher")))
-        .andExpect(jsonPath("$._embedded.posts[1].id", is("post-2-researcher")))
-        .andExpect(jsonPath("$._embedded.posts[2].id", is("post-3-researcher")))
-        .andExpect(jsonPath("$._embedded.posts[3].id", is("post-1-representative")))
-        .andExpect(jsonPath("$._embedded.posts[4].id", is("post-2-representative")))
-        .andExpect(jsonPath("$._embedded.posts[5].id", is("post-3-representative")))
-        .andExpect(jsonPath("$._embedded.posts[6].id", is("post-4-representative")));
+        .andExpect(
```
Comment (svituz) [inline src/test/java/eu/bbmri_eric/negotiator/integration/api/v3/PostControllerTests.java:157]: shouldn't the posts returned in a specific order?

File: src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java (hunk 3 above)
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java:25]: why? Don't we want to catch all exceptions? the result is the same, no authenticated user can be parsed?

Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java:25]: https://stackoverflow.com/questions/21938/is-it-really-that-bad-to-catch-a-general-exception

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java (hunk 5 above)
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java:268]: hmm... this is applicable only for networks, so maybe either leave it as it is or create a NetworkAccessManger?

File: src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java (hunk 3 above)
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java:27]: added logging

**Review by svituz — APPROVED** [review]

**Pull request merged**